python/ray/train/_internal/state/state_manager.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import os
+from typing import Dict
+
+import ray
+from ray.data import Dataset
+from ray.train._internal.state.schema import (
+    TrainDatasetInfo,
+    TrainRunInfo,
+    TrainWorkerInfo,
+)
+from ray.train._internal.state.state_actor import (
+    TRAIN_STATE_ACTOR_NAME,
+    TRAIN_STATE_ACTOR_NAMESPACE,
+)
+from ray.train._internal.utils import check_for_failure
+from ray.train._internal.worker_group import WorkerGroup
+
+logger = logging.getLogger(__name__)
+
+
+class TrainRunStateManager:
+    """A class that aggregates and reports train run info to TrainStateActor.
+
+    This manager class is created on the train controller layer for each run.
+    """
+
+    def __init__(self, worker_group: WorkerGroup) -> None:
+        self.state_actor = ray.get_actor(
+            name=TRAIN_STATE_ACTOR_NAME, namespace=TRAIN_STATE_ACTOR_NAMESPACE
+        )
```

Comment:
This manager will raise an error if the actor hasn't been created. Should we make this a clearer condition by making `state_actor` an argument to pass into the constructor?

So the person constructing the state manager must ensure that the actor exists before creating this manager. I think it will be better for unit tests too.

Reply:
That's a great idea! The creator should be responsible of creating the StateActor.